### ISSUE TITLE
The calendar dialog is missing an accessibility property

### ISSUE THREAD

**Issue by barettjfed** [Hacker0x01/react-datepicker#4555]
**Describe the bug**
According to the ARIA authoring practices guide the dialog should contain an aria-label. [Link](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/#:~:text=dialog%20is%20modal.-,aria%2Dlabel%3D%22string%22,-div)

**To Reproduce**

1. Open up calendar
2. Notice there is no aria-label on the aria-modal

**Expected behavior**
aria-label should be set on the aria-modal dialog to something like "choose date".

**Screenshots**
![image](https://github.com/Hacker0x01/react-datepicker/assets/125996778/6c785029-0a32-4781-9ab6-c604cdef569c)

**Additional context**
This is our last open issue from our accessibility vendor on our date picker component. Good job!
